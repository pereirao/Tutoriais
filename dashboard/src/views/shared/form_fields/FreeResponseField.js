// @flow

import React, {PureComponent} from 'react';
import I18n from '../../../localization';
import style_sheet from './form_fields.module.css';

type FreeResponseFieldProps = { onChangeText: (string, Object) => void, is_required: boolean };

class FreeResponseField extends PureComponent<FreeResponseFieldProps> {
  props: FreeResponseFieldProps

  static defaultProps = {name: 'notes', is_required: true}
  input: ReactClass<*>

  getValidity = value => (
    this.props.is_required ? !!value : true
  )

  onChangeText = value => {
    this.props.onChangeText(value, this.getValidity(value));
  }

  handleChange = ({target}) => {
    this.onChangeText(target.value);
  }

  render() {
    return (
      <textarea
        placeholder={I18n.t(`form.placeholder.${this.props.name}`)}
        ref={el => {
          this.input = el;
        }}
        className={style_sheet.textArea}
        {...this.props}
        onChange={this.handleChange}/>
    );
  }
}

export default FreeResponseField;
