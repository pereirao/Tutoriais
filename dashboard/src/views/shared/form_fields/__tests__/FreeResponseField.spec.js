import React from 'react';

import FreeResponseField from '../FreeResponseField';

it('renders correctly', () => {
  expect(shallow(<FreeResponseField />)).toMatchSnapshot();
});
