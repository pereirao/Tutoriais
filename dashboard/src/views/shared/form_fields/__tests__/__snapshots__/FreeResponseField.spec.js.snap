// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<textarea
  className="textArea"
  is_required={true}
  name="notes"
  onChange={[Function]}
  placeholder="Notes"
/>
`;
